drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -286,10 +305,13 @@ protected void checkTypeSupport(DMNType type,
                 if (alreadyInCollection && nestedType.isComposite()) {
                     errorHolder.getMultipleNestedObject().add(nestedPath);
                 }
-                checkTypeSupport(nestedType,
-                                 alreadyInCollection || nestedType.isCollection(),
-                                 errorHolder,
-                                 nestedPath);
+                if (!alreadyVisited.contains(nestedType.getName())) {
+                    internalCheckTypeSupport(nestedType,
+                                             alreadyInCollection || nestedType.isCollection(),
```

Comment:
@danielezonca 
If _alreadyInCollection_  is a specification of _alreadyVisited_, would not be possible to merge them, maybe as a map where value is a boolean to specify it is a collection or not?

Reply:
They are not the same, `alreadyInCollection` is related to check type support while `alreadyVisited` is to avoid infinite recursion.
I.e. a structure like `List[Person[List[PhoneNumber]]` is not valid from a check type support so `alreadyInCollection` will be `true` during the analysis at a certain point and this information cannot be preserved just considering `alreadyVisited`